### PR TITLE
Detect gsed and use it if it exists

### PR DESCRIPTION
**Pull request by anshul**

### CODE REVIEW

**Comment by anshul** [comment]
@stevemao Squashed the changes into a single commit.

**Comment by stevemao** [comment]
Is it common to symlink `gsed` to `sed`? homebrew does it. Maybe we should instead warn the user?

**Comment by anshul** [comment]
This is a common pattern on osx/homebrew.   `brew install sed find readlink` will create `gsed`, `gfind`, `greadlink`.  Overriding the apple implementations of these commands can give unpredictable behaviour for various apple software.  I never use or recommend `--with-default-names`

**Comment by anshul** [comment]
http://apple.stackexchange.com/questions/69223/how-to-replace-mac-os-x-utilities-with-gnu-core-utilities

**Comment by anshul** [comment]
Sorry, I read your comment wrong.  Edited mine.  I think, everyone is served better if npm install just works out of the box on a typical osx install.  Recommending overriding the apple sed command could leave other commands that depend on the apple installation in a bad shape.

**Comment by anshul** [comment]
Looks like ports also installs a gsed. http://apple.stackexchange.com/a/166593

**Comment by stevemao** [comment]
I'm fine with this. I'll see what other people think.

**Comment by paulirish** [comment]
Yup. This is good. Osx sed is crap but some people won't want to replace it and that's fine.

File: diff-so-fancy [hunk 1]
```diff
@@ -3,6 +3,10 @@
 [ $# -ge 1 -a -f "$1" ] && input="$1" || input="-"
 
 diff_highlight="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/third_party/diff-highlight/diff-highlight"
+SED=sed
+if hash gsed 2>/dev/null; then
+  SED=gsed
+fi
```
Comment (paulirish) [inline diff-so-fancy]: Can you make this a one liner?

Comment (anshul) [inline diff-so-fancy]: Sure

**Comment by anshul** [comment]
@paulirish Does that one liner look fine?

**Comment by paulirish** [comment]
Yup! Ace.

**Pull request merged**